### PR TITLE
Tng format

### LINKED ISSUE #865: Support for Gromacs's TNG format
Gromacs is slowly moving toward the new TNG trajectory format. The format got introduced in the late releases of the 4 serie, and should become the default format eventually.

TNG allows better compression that XTC, and have random frame access built in. It also allows to store arbitrary data along side the trajectory (which may require #785), and time-dependant topology (which would require #864).

A library is provided by the gromacs people: https://github.com/gromacs/gromacs/tree/master/src/external/tng_io

See the paper: http://onlinelibrary.wiley.com/doi/10.1002/jcc.23495/full

Full support for the TNG format can be implemented in parts:

* [ ] read trajectory with constant topology
* [ ] write trajectory with constant topology
* [ ] read constant topology
* [ ] read trajectory with time-dependant topology
* [ ] write trajectory with time-dependant topology
* [ ] read arbitrary data
* [ ] write arbitrary data

### PR DESCRIPTION
**Pull request by kain88-de**
Fixes #865

First stumbs of support for the tng format based on [pytng](https://github.com/MDAnalysis/pytng)

@richardjgowers I don't know how the topology information has to be structured. If you could give me some hints I'll add it too pytng. 

The coordinate reader should work but I have to add tests.

PR Checklist
------------
 - [ ] Tests?
 - [ ] Docs?
 - [ ] CHANGELOG updated?
 - [ ] Issue raised/referenced?

### CODE REVIEW

File: package/MDAnalysis/coordinates/TNG.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+    @property
+    def n_frames(self):
+        """number of frames in trajectory"""
+        return len(self._tng)
+
+    def _reopen(self):
+        """reopen trajectory"""
+        self.ts.frame = 0
+        self._frame = -1
```
Comment (richardjgowers) [inline package/MDAnalysis/coordinates/TNG.py]: Why do we need '_frame'? We can just use ts.frame

**Comment by richardjgowers** [comment]
@kain88-de I guess with the tests we want to use the adk.trr tests we already have? Trjconv worked on the whole trajectory iirc

**Comment by orbeckst** [comment]
Great stuff!

The only thing is that technically it cannot go into 0.16.1 – it's a new feature. So, I suggest, we get 0.16.1 out asap (which also gives `pytng` a little bit of time to get on conda) and then we can do a 0.17.0 pretty soon afterwards.

File: package/MDAnalysis/coordinates/TNG.py (hunk 1 above)
Comment (kain88-de) [inline package/MDAnalysis/coordinates/TNG.py]: I'll look that up again. There might be some offset issues here.

**Comment by kain88-de** [comment]
> @kain88-de I guess with the tests we want to use the adk.trr tests we already have? Trjconv worked on the whole trajectory iirc

yes that would be good. Then I can copy/paste the tests

File: testsuite/MDAnalysisTests/datafiles.py [hunk 2]
```diff
@@ -136,6 +136,7 @@
     "Martini_membrane_gro", # for testing the leaflet finder
     "COORDINATES_XTC",
     "COORDINATES_TRR",
+    "COORDINATES_TNG",
```
Comment (jbarnoud) [inline testsuite/MDAnalysisTests/datafiles.py]: The TNG format is planned to evolve (arbitrary data, new compression algorithms...), a comment saying how the file was produced, and with what version of gromacs would be beneficial I think.

File: package/MDAnalysis/topology/TNGParser.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+     - names
+     - chainids
+     - resids
+     - resnames
+
+    Guesses the following Attributes:
+     - masses
+
+
+    .. versionadded:: 0.16.1
```
Comment (jbarnoud) [inline package/MDAnalysis/topology/TNGParser.py]: This will have to be updated.

**Comment by orbeckst** [comment]
Unlikely to happen for 0.17.0 – kicked down the road.

**Comment by kain88-de** [comment]
I will not be able to finish this. Someone else can pick it up if they like. I will leave the branch on the repository.

**Pull request closed without merging**